src/mumble/Messages.cpp

```diff
@@ -888,30 +887,32 @@ void MainWindow::msgTextMessage(const MumbleProto::TextMessage &msg) {
 	bool privateMessage = false;
 
 	if (msg.tree_id_size() > 0) {
-		target += tr("(Tree) ");
+		target += tr("Tree");
 	} else if (msg.channel_id_size() > 0) {
-		target += tr("(Channel) ");
+		target += tr("Channel");
 	} else if (msg.session_size() > 0) {
-		target += tr("(Private) ");
+		target += tr("Private");
 		privateMessage = true;
 	}
 
 	// If NoScope or NoAuthor is selected generate a new string to pass to TTS
-	if (g.s.bTTSNoScope || g.s.bTTSNoAuthor) {
+	const QString overrideTTS = [&]() {
+		if (!g.s.bTTSNoScope && !g.s.bTTSNoAuthor) {
+			return QString();
+		}
+		const QString plain = QTextDocumentFragment::fromHtml(u8(msg.message())).toPlainText();
 		if (g.s.bTTSNoScope && g.s.bTTSNoAuthor) {
-			overrideTTS += tr("%2%1: %3").arg(QString()).arg(QString()).arg(u8(msg.message()));
-		} else if (g.s.bTTSNoAuthor) {
-			overrideTTS += tr("%2%1: %3").arg(QString()).arg(target).arg(u8(msg.message()));
-		} else if (g.s.bTTSNoScope) {
-			overrideTTS += tr("%2%1: %3").arg(name).arg(QString()).arg(u8(msg.message()));
+			return plain;
 		}
-	}
+		const QString prefix = g.s.bTTSNoScope ? plainName : target;
+		return tr("%1: %2").arg(prefix).arg(plain);
+	}();
 
 	g.l->log(privateMessage ? Log::PrivateTextMessage : Log::TextMessage,
-	         tr("%2%1: %3").arg(name).arg(target).arg(u8(msg.message())),
+	         tr("(%2) %1: %3").arg(name).arg(target).arg(u8(msg.message())),
```

Comment:
If target were empty this would lead to wrong formatting. Could this ever be the case?

Reply:
I think atm this is not possible, but maybe this could change in the future. Thus I'd say it's probably better to check for this case explicitly than implicitly relying on it to never occur ^^

Reply:
Ok, I made some changes to account for that.